projects/dashboards/src/widgets/greeting-label/greeting-label-widget-renderer.component.test.ts

```diff
@@ -0,0 +1,38 @@
+import { GreetingLabelComponent } from '@hypertrace/components';
+import { RENDERER_API } from '@hypertrace/hyperdash-angular';
+import { createComponentFactory, Spectator } from '@ngneat/spectator/jest';
+import { EMPTY } from 'rxjs';
+import { GreetingLabelWidgetRendererComponent } from './greeting-label-widget-renderer.component';
+import { GreetingLabelWidgetModel } from './greeting-label-widget.model';
+
+describe('Greeting label widget renderer component', () => {
+  let spectator: Spectator<GreetingLabelWidgetRendererComponent>;
+  const mockModel: Partial<GreetingLabelWidgetModel> = {};
+  const createComponent = createComponentFactory<GreetingLabelWidgetRendererComponent>({
+    component: GreetingLabelWidgetRendererComponent,
+    entryComponents: [GreetingLabelComponent],
+    providers: [
+      {
+        provide: RENDERER_API,
+        useFactory: () => ({
+          model: mockModel,
+          getDataFromModelDataSource: EMPTY,
+          getTimeRange: jest.fn(),
+          change$: EMPTY,
+          dataRefresh$: EMPTY,
+          timeRangeChanged$: EMPTY
+        })
+      }
+    ]
+  });
+
+  beforeEach(() => {
+    mockModel.labelTemplate = '{greeting}, test';
+    spectator = createComponent();
+  });
+
+  test('should render greeting label correctly', () => {
+    spyOn(Date.prototype, 'getHours').and.returnValue(9);
```

Comment:
I'm confused on this test - shouldn't 9 be morning? I'm assuming this is because the greeting message is already set before we spied here (it'd be set on 31), so the spy has no effect. But I don't think that's the intention here, and if it is A. the test isn't clear on it and B. that would make this test produce different results at different times of day which is of course, not good.

Reply:
That was a mistake. Fixed it.